betty/json/schema.py

```diff
@@ -71,10 +84,17 @@ def validate(self, data: Any) -> None:
         """
         Validate data against this schema.
         """
-        schema = Resource.from_contents(self.schema)
-        schema_registry = schema @ Registry()  # type: ignore[operator, var-annotated]
+        schema = expand_schema_names(self.schema)
+        if "$id" not in schema:
+            # Set a dummy $id to satisfy the registry. This is only ever needed for unnamed
+            # schemas.
+            schema = {
+                "$id": "betty",
+                **schema,
+            }
+        schema_registry = Resource.from_contents(schema) @ Registry()  # type: ignore[operator, var-annotated]
         validator = Draft202012Validator(
```

Comment:
Use the `validate()` API function and let the API determine the correct validator?

Reply:
No, the entire API assumes this dialect so let's leave it for now until (if ever) it's actually pluggable.